File: DataStructuresAndLibraries/pagina03/1709/1709.lua
-- Calcula a nova posição que a carta terá após um embaralhamento.
local function new_position(pos, n)
    local mid = math.ceil(n / 2)
    if pos < mid then return pos * 2 + 1 end
    return (pos - mid) * 2
end

local n = tonumber(io.read())
local pos = 1
local steps = 1

local aux = new_position(pos, n)
while aux ~= pos do
    steps = steps + 1
    aux = new_position(aux, n)
end

print(steps)

File: DataStructuresAndLibraries/pagina03/1709/1709.cpp
#include <iostream>
#include <cmath>
using namespace std;

int new_position(int pos, int n){
    int mid = ceil(n / 2);
    if(pos < mid) return pos * 2 + 1;
    return (pos - mid) * 2;
}

int main() {
    int n;
    int steps = 1, pos = 1;
    cin >> n;

    int aux = new_position(pos, n);
    while(aux != pos){
        steps++;
        aux = new_position(aux, n);
    }

    cout << steps << "\n";
    return 0;
}